CMakeLists.txt

```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
 
-set(version "1.4.0" CACHE STRING "Project version")
-
-string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]" version_short ${version})
+set(BUILD_NUMBER "0" CACHE STRING "Build number")
```

Comment:
Please overtake the variant we settled on in Nik's PR :point_up:

Reply:
`build-number`, right?

Reply:
As discussed previously, I think the way it is represented here may be more correct.

Reply:
> build-number, right?

Are you referring to the spelling? That's actually not what I was aiming at with my comment. I was referring to the whole cmake code that was used to introduce the variable in Nik's PR (only if `packaging=ON`, default to empty string and error if it is empty but `packaging=ON`)

Reply:
> I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

So, like this `BUILD_NUMBER`. But I think that options should probably be in this syntax and variables in lower case with `-` or `_`. It would make identifying the types easier. Or the other way if that is more helpful.

Reply:
I don't think using the case of the names as a means to express the variable's type is a good thing :thinking:

Reply:
The proper description is still missing. Could you please overtake that from Nik's PR?